pkg/codegen/pcl/binder.go

```diff
@@ -256,6 +272,33 @@ func (b *binder) declareNodes(file *syntax.File) (hcl.Diagnostics, error) {
 	return diagnostics, nil
 }
 
+// Pop an attribute that's internal to Pulumi, e.g.: "__uniqueName" on a resource or output.
+func popStringAttr(attrs hclsyntax.Attributes, key string) (string, *hcl.Diagnostic) {
```

Comment:
Is this how we handle other pulumi special attributes? I'm concerned about destructively mutating the program as we bind it. Do we mention somewhere that a program cannot be bound twice?

Reply:
I couldn't find anywhere else where we used a side channel like this, like `__default`. I think that's a good point about mutating the program. Will look a bit deeper.

Agree on the renames.

Reply:
https://github.com/pulumi/pulumi/issues/9469